### ISSUE TITLE
cpu run time

### ISSUE THREAD

**Issue by zhangyunming** [cuiziteng/Illumination-Adaptive-Transformer#13]
hi,  i test the model on my win7 pc i7 cpu,

the time is 13s?  it is normal?
1it [00:12, 12.53s/it]

LOL_V2

**Comment by cuiziteng** [comment]
I think it's not normal, the results in paper is running on 3090 GPU, beyond the different device, maybe the ssim/psnr calculation in CPU take too much time? I think maybe you could check again.

**Comment by zhangyunming** [comment]
the code is:
import torch
import torch.nn as nn
import torchvision
import torch.backends.cudnn as cudnn
import torch.optim
import torch.nn.functional as F
import time 

import os
import argparse
import numpy as np
from utils import PSNR, validation, LossNetwork
from model.IAT_main import IAT
from IQA_pytorch import SSIM, MS_SSIM
from data_loaders.lol import lowlight_loader
from tqdm import tqdm

parser = argparse.ArgumentParser()
#parser.add_argument('--gpu_id', type=str, default=0)
parser.add_argument('--save', type=bool, default=True)
parser.add_argument('--img_val_path', type=str, default="F:\\zym_live_HDR\\LOL_v2\\LOL_v2\\Test\\Low\\")
parser.add_argument('--pre_norm', type=bool, default=True)
config = parser.parse_args()

print(config)
val_dataset = lowlight_loader(images_path=config.img_val_path, mode='test', normalize=config.pre_norm)
val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=1, shuffle=False)
#os.environ['CUDA_VISIBLE_DEVICES'] = str(config.gpu_id)

model = IAT().cpu()
model.load_state_dict(torch.load("best_Epoch_lol_v2.pth",map_location='cpu'))
model.eval()


ssim = SSIM()
psnr = PSNR()
ssim_list = []
psnr_list = []

def mkdir(path):
    if not os.path.exists(path):
        os.mkdir(path)

if config.save:
    result_path = config.img_val_path.replace('Low', 'Result')
    mkdir(result_path)

with torch.no_grad():
    for i, imgs in tqdm(enumerate(val_loader)):
        #print(i)
        low_img, high_img = imgs[0].cpu(), imgs[1].cpu()
        #print(low_img.shape)
        
        t_begin = time.time()
        mul, add ,enhanced_img = model(low_img)
        t_end = time.time()
        print("the time is: ",(t_end-t_begin))

        torchvision.utils.save_image(enhanced_img, result_path + str(i) + '.png')

        ssim_value = ssim(enhanced_img, high_img, as_loss=False).item()
        psnr_value = psnr(enhanced_img, high_img).item()

        ssim_list.append(ssim_value)
        psnr_list.append(psnr_value)


SSIM_mean = np.mean(ssim_list)
PSNR_mean = np.mean(psnr_list)
print('The SSIM Value is:', SSIM_mean)
print('The PSNR Value is:', PSNR_mean)



and the time is: 12.xx s

**Comment by cuiziteng** [comment]
那确实不太正常，是不是transformer结构在CPU跑的慢呢，或者torch的transformer在CPU上高效运转有关？这方面我就不是很清楚了，您有GPU的话可以测试GPU速度，并且建议多张图像一起跑，算平均速度。

**Comment by zhangyunming** [comment]
# Short Cut Connection on Final Layer
class Local_pred_S(nn.Module):
    def __init__(self, in_dim=3, dim=16, number=4, type='ccc'):
        super(Local_pred_S, self).__init__()
  
    def forward(self, img):
        img1 = self.relu(self.conv1(img))
        # short cut connection
        t_begin = time.time()
        mul = self.mul_blocks(img1) + img1
        t_end = time.time()
        add = self.add_blocks(img1) + img1
        mul = self.mul_end(mul)
        add = self.add_end(add)
        
        print("the time mul is: ",(t_end-t_begin))
       

 主要耗时在这里  
        t_begin = time.time()
        mul = self.mul_blocks(img1) + img1
        t_end = time.time()

每个6s，看起来网络不复杂啊，我需要整到手机上，所以就不测N卡的速度了

**Comment by cuiziteng** [comment]
那些是我们的早期设想模块，代码直接加上了，最后implenment的时候没有加上，有兴趣的话可以自己尝试～
